dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/SolrSearchResultsService.java

```diff
@@ -0,0 +1,327 @@
+
+    private static final String DATAVERSES_QUERY_BASE_NAME = "Dataverse.findDataForSolrResults";
+
+    private static final int DATAVERSES_QUERY_ID = 0;
+    private static final int DATAVERSES_QUERY_AFFILIATION = 1;
+    private static final int DATAVERSES_QUERY_ALIAS = 2;
+    private static final int DATAVERSES_QUERY_PARENT_ALIAS = 3;
+
+    private static final String DATASETS_QUERY_BASE_NAME = "Dataset.findDataForSolrResults";
+
+    private static final int DATASETS_QUERY_VERSION_ID = 0;
+    private static final int DATASETS_QUERY_VERSION_STATE = 1;
+    private static final int DATASETS_QUERY_DATAVERSE_ALIAS = 2;
+    private static final int DATASETS_QUERY_THUMBNAIL_ID = 3;
+    private static final int DATASETS_QUERY_USE_GENERIC_THUMBNAIL = 4;
+    private static final int DATASETS_QUERY_STORAGE_ID = 5;
+
+    private static final String DATAFILES_QUERY_BASE_NAME = "Datafile.findDataForSolrResults";
+
+    private static final int DATAFILES_QUERY_FILE_ID = 0;
+    private static final int DATAFILES_QUERY_FILE_CREATEDATE = 1;
+    private static final int DATAFILES_QUERY_FILE_INDEXTIME = 2;
+    private static final int DATAFILES_QUERY_FILE_MODIFICATIONTIME = 3;
+    private static final int DATAFILES_QUERY_FILE_PERMISSIONINDEXTIME = 4;
+    private static final int DATAFILES_QUERY_FILE_PERMISSIONMODIFICATIONTIME = 5;
+    private static final int DATAFILES_QUERY_FILE_PUBLICATIONDATE = 6;
+    private static final int DATAFILES_QUERY_FILE_PREVIEWIMAGEAVAILABLE = 7;
+    private static final int DATAFILES_QUERY_FILE_STORAGEIDENTIFIER = 8;
+    private static final int DATAFILES_QUERY_FILE_AUTHORITY = 9;
+    private static final int DATAFILES_QUERY_FILE_PROTOCOL = 10;
+    private static final int DATAFILES_QUERY_FILE_IDENTIFIER = 11;
+    private static final int DATAFILES_QUERY_FILE_CONTENTTYPE = 12;
+    private static final int DATAFILES_QUERY_FILE_FILESIZE = 13;
+    private static final int DATAFILES_QUERY_FILE_INGESTSTATUS = 14;
+    private static final int DATAFILES_QUERY_FILE_CHECKSUMVALUE = 15;
+    private static final int DATAFILES_QUERY_FILE_CHECKSUMTYPE = 16;
+    private static final int DATAFILES_QUERY_FILE_PREVIOUSDATAFILEID = 17;
+    private static final int DATAFILES_QUERY_FILE_ROOTDATAFILEID = 18;
+    private static final int DATAFILES_QUERY_DATASET_ID = 19;
+    private static final int DATAFILES_QUERY_DATASET_IDENTIFIER = 20;
+    private static final int DATAFILES_QUERY_DATASET_AUTHORITY = 21;
+    private static final int DATAFILES_QUERY_DATATABLE_ID = 22;
+    private static final int DATAFILES_QUERY_DATATABLE_UNF = 23;
+    private static final int DATAFILES_QUERY_DATATABLE_CASEQUANTITY = 24;
+    private static final int DATAFILES_QUERY_DATATABLE_VARQUANTITY = 25;
+    private static final int DATAFILES_QUERY_DATATABLE_ORIGINALFILEFORMAT = 26;
+    private static final int DATAFILES_QUERY_DATATABLE_ORIGINALFILESIZE = 27;
```

Comment:
PREVIOUSDATAFILEID, ROOTDATAFILEID, ORIGINALFILEFORMAT, ORIGINALFILESIZE are also not used